src/repo/git.go

```diff
@@ -43,56 +49,144 @@ func (g *Git) Clone(repo domain.Repository, accessToken string, branch, sha stri
 		u.User = url.UserPassword("x-access-token", accessToken)
 	}
 
+	// Create a progress writer using the deployment ID from the repository
+	// Using the repository TreenqID as the deployment ID
+	progressWriter := g.progressBuf.AsWriter(fmt.Sprintf("repo-%s", dir), slog.LevelInfo)
+	g.progressBuf.Append(fmt.Sprintf("repo-%s", dir), domain.ProgressMessage{
+		Payload: fmt.Sprintf("Cloning repository from %s", repo.CloneURL()),
+		Level:   slog.LevelInfo,
+	})
+
 	r, err := git.PlainClone(dir, false, &git.CloneOptions{
 		URL:      u.String(),
-		Progress: os.Stdout,
+		Progress: progressWriter,
```

Comment:
I this component I meant only this.
the rest is written from outside on error or success.
as I shared in docker build example, it has no event inside the build statement